Implement a more stylable axes

rickimaru18 opened the pull request:
# 📲 What #

* Support a more customizable axes.

# 🤔 Why #

* New feature.

# 🛠 How #

* Add style class which contains all the styling property for X and Y axes.

# ✅ Acceptance criteria #

- [x] Passing Unit Tests
- [x] Passing UI Tests

# ⏰ TODO #

* ~~Font styles for axes values~~

# 🔗 References #

<img width="310" alt="image" src="https://user-images.githubusercontent.com/13028267/119943945-4232fe00-bfc6-11eb-861c-fa76f9628841.png">

rickimaru18 committed:
Implement a more stylable axes

# Add Style classes
- Use Style class to set color, visibility, stroke width and style.
- Use AxesStyle when styling X and Y axes.

# In Chart classes
- Replace axes styling properties to AxesStyle. (Chart, LineChart, BarChart)
- Draw axes above the chart data. (LineChart, BarChart)

# In sample Fragments
- Use "axesStyle" property.
- Update data color.

# In UI test
- Use "axesStyle" property.

rexendy reviewed (changes requested):
Rick pls add all the colors to colors.xml for easy tracking and maintenance

Thank you

rickimaru18 committed:
Use resource colors

# Add resource colors
- "gray" used for axes
- "green" used for budget chart data
- "red" used for spent chart data

# In sample Fragments
- Use resource colors.
- Update viewmodels to initialize data and accept colors from resource.

rickimaru18 committed:
Support axes values styling

# In Style
- Add FontStyle which contains the properties for font styling.
- In AxesStyle, support axis value styling.

# In Chart::createAxisTextPaint(...)
- Refer to axis' "fontStyle" property to set Paint's text styling.

# In sample Fragments
- Add X-axis font styling.

rickimaru18 committed:
Use old API for getting resource color.

rickimaru18 committed:
Use ResourceCompat when getting resource color.
